### ISSUE TITLE
no screenshots occurring

### ISSUE THREAD

**Issue by hjbockholt** [reuter-lab/qatools-python#25]
## Question/Support Request

the screenshots fail when I try to run a simple single subject toy example

## Screenshots

Starting qatools-python for subject 013_S_4580_S680266 at 2022-09-15 21:47 UTC

Checking size of the corpus callosum ...
Relative size of the corpus callosum is 0.001738
Checking topology of the surfaces ...
Number of defects in the left hemisphere: 60
Topological fixing time for the left hemisphere: 9.1 min
Number of defects in the right hemisphere: 79
Topological fixing time for the right hemisphere: 33.7 min
Number of holes in the left hemisphere: 157
Number of holes in the right hemisphere: 122
Checking WM/GM contrast SNR ...
WM/GM contrast SNR for the left hemisphere: 2.22
WM/GM contrast SNR for the right hemisphere: 2.277
Computing Talairach rotation angles ...
Found Talairach rotation angles: x = -0.0858 , y = -0.0176 , z = -0.00875 radians.

-----------------------------
Creating screenshots ...

Using default for screenshot base image
Using default for screenshot overlay image
Using default for screenshot surface
**ERROR: screenshots module failed for subject 013_S_4580_S680266**
Finished subject 013_S_4580_S680266 at 2022-09-15 21:47 UTC

**also:**
venv) [jbockholt@arclogin01 scripts]$ more error_1509480.err
/usr/bin/lua: /usr/share/lmod/lmod/libexec/MT.lua:1420: attempt to compare string with nil
stack traceback:
	/usr/share/lmod/lmod/libexec/MT.lua:1420: in function </usr/share/lmod/lmod/libexec/MT.lua:1418>
	[C]: in function 'table.sort'
	/usr/share/lmod/lmod/libexec/MT.lua:1418: in function 'MT.list'
	/usr/share/lmod/lmod/libexec/Master.lua:616: in function 'Master.unload'
	/usr/share/lmod/lmod/libexec/MasterControl.lua:370: in function 'MasterControl.unload'
	/usr/share/lmod/lmod/libexec/Master.lua:409: in function 'Master.load'
	/usr/share/lmod/lmod/libexec/MasterControl.lua:285: in function 'MasterControl.load'
	/usr/share/lmod/lmod/libexec/Master.lua:555: in function 'Master.reloadAll'
	/usr/share/lmod/lmod/libexec/cmdfuncs.lua:928: in function 'Use'
	/usr/share/lmod/lmod/libexec/lmod:604: in function 'main'
	/usr/share/lmod/lmod/libexec/lmod:650: in main chunk
	[C]: in ?
/usr/bin/lua: /usr/share/lmod/lmod/libexec/MT.lua:1420: attempt to compare string with nil
stack traceback:
	/usr/share/lmod/lmod/libexec/MT.lua:1420: in function </usr/share/lmod/lmod/libexec/MT.lua:1418>
	[C]: in function 'table.sort'
	/usr/share/lmod/lmod/libexec/MT.lua:1418: in function 'MT.list'
	/usr/share/lmod/lmod/libexec/Master.lua:616: in function 'Master.unload'
	/usr/share/lmod/lmod/libexec/MasterControl.lua:370: in function 'MC_Load.unload'
	/usr/share/lmod/lmod/libexec/Master.lua:409: in function 'Master.load'
	/usr/share/lmod/lmod/libexec/MasterControl.lua:285: in function 'MC_Load.load'
	/usr/share/lmod/lmod/libexec/MasterControl.lua:264: in function 'MC_Load.load_usr'
	/usr/share/lmod/lmod/libexec/cmdfuncs.lua:423: in function 'Load_Usr'
	/usr/share/lmod/lmod/libexec/lmod:604: in function 'main'
	/usr/share/lmod/lmod/libexec/lmod:650: in main chunk
	[C]: in ?
/home/users/jbockholt/src/qatools-python/qatoolspython/qatoolspython.py:1080: FutureWarning: The error_bad_lines argument has been deprecated and will be removed in a future version.

  createScreenshots(SUBJECT=subject, SUBJECTS_DIR=subjects_dir, OUTFILE=outfile, INTERACTIVE=False, BASE=screenshots_base_subj, OVERLAY=screenshots_overlay_subj, SURF=screenshots_surf_subj, VIEWS=screenshots_views, LAYOUT=screenshots_layout)
/home/users/jbockholt/src/qatools-python/qatoolspython/qatoolspython.py:1080: FutureWarning: The warn_bad_lines argument has been deprecated and will be removed in a future version.

  createScreenshots(SUBJECT=subject, SUBJECTS_DIR=subjects_dir, OUTFILE=outfile, INTERACTIVE=False, BASE=screenshots_base_subj, OVERLAY=screenshots_overlay_subj, SURF=screenshots_surf_subj, VIEWS=screenshots_views, LAYOUT=screenshots_layout)

## Environment
 - qatools-python Version: Python 3.9.9 using qatools-python **Update v1.2**

 - OS: ...NAME="CentOS Linux"
VERSION="7 (Core)"

 - CPU: (venv) [jbockholt@arclogin01 scripts]$ cat /proc/cpuinfo 
processor	: 0
vendor_id	: GenuineIntel
cpu family	: 15
model		: 6
model name	: Common KVM processor
stepping	: 1
microcode	: 0x1
cpu MHz		: 2394.380
cache size	: 16384 KB
physical id	: 0
siblings	: 1
core id		: 0
cpu cores	: 1
apicid		: 0
initial apicid	: 0
fpu		: yes
fpu_exception	: yes
cpuid level	: 13
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 syscall nx lm constant_tsc nopl xtopology eagerfpu pni cx16 x2apic hypervisor lahf_lm
bogomips	: 4788.76
clflush size	: 64
cache_alignment	: 128
address sizes	: 40 bits physical, 48 bits virtual
power management:

(venv) [jbockholt@arclogin01 scripts]$ pip freeze
brainprint==0.2
contourpy==1.0.5
cycler==0.11.0
fonttools==4.37.2
imageio==2.21.3
kiwisolver==1.3.2
LaPy==0.3
matplotlib==3.4.3
networkx==2.8.6
nibabel==3.2.1
numpy==1.22.0
packaging==21.2
pandas==1.3.4
Pillow==9.0.1
plotly==5.10.0
pyparsing==2.4.7
python-dateutil==2.8.2
pytz==2021.3
PyWavelets==1.4.0
qatoolspython==1.3.0
scikit-image==0.19.3
scipy==1.7.2
six==1.16.0
tenacity==8.0.1
tifffile==2022.8.12
transforms3d==0.3.1

### Execution

Run Command: 
python3 $qatools_dir/qatools.py --subjects_dir $SUBJECTS_DIR --output_dir $qa_dir --subjects 013_S_4580_S680266 --screenshots --screenshots-html

**Comment by kdiers** [comment]
Hi, and thanks for the error report. We'll look into this and get back very soon.

**Comment by kdiers** [comment]
Hi, and thanks again for your message. I believe I have identified the problem, and can likely provide a fix within the next days. In the meantime and as a workaround, you could switch off part of the screenshots generation (ie. surfaces / surface overlays) by adding the `--screenshots_surf none` option to your command.

**Comment by kdiers** [comment]
Hi, here's a quick update on our search for the issue with running the screenshots module:

- I thought that I had reproduced the issue when I downloaded the scripts into a new environment and did a manual install of the dependencies (without specifying their version numbers). That gave the same error as reported above. Reinstalling the dependencies by using the `requirements.txt` file (which includes version numbers) solved the issue, i.e. the script ran through properly. I could narrow down the issue to the `scipy` package, where the current version from the manual install (1.9.1) provokes the issue, and switching back to version 1.7.2 as specified in the requirements file solves the issue. 

- However, although the symptoms are the same, I doubt that this is the cause of the problem in your case, since your `pip freeze` indicates that you're also using scipy 1.7.2. Also, using exactly the same package versions as reported above (which are slightly different from the requirements) lets the program run smoothly at my site. So there is probably more than one cause for the issue.

- One thing to consider is the possibility that the issue is related to the specific dataset that you used for testing. Could you possibly do a test with the `bert` dataset that is included in every FreeSurfer distribution (in the `subjects` directory) - which is what I did for testing - and let us know the result?

**Comment by alx080** [comment]
Hi, running QAtools on a fast surfer subject I'm getting this error when --screenshots is activated:

```
Creating screenshots ...

Using default for screenshot base image
Using default for screenshot overlay image
Using default for screenshot surface
/home/ubuntu/qatools-python/qatoolspython/createScreenshots.py:112: FutureWarning: The error_bad_lines argument has been deprecated and will be removed in a future version. Use on_bad_lines in the future.


  lut = pd.read_csv(os.path.join(os.environ['FREESURFER_HOME'],'FreeSurferColorLUT.txt'),
/home/ubuntu/qatools-python/qatoolspython/createScreenshots.py:112: FutureWarning: The warn_bad_lines argument has been deprecated and will be removed in a future version. Use on_bad_lines in the future.


  lut = pd.read_csv(os.path.join(os.environ['FREESURFER_HOME'],'FreeSurferColorLUT.txt'),
/home/ubuntu/.local/lib/python3.8/site-packages/numpy/lib/arraysetops.py:738: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.
  element = np.asarray(element)
ERROR: screenshots module failed for subject ID1postOP1
Finished subject ID1postOP1 at 2022-11-12 17:26 UTC

```
Numpy 1.23/1.22
Your `--screenshots_surf none` workaround works!

**Comment by m-reuter** [comment]
Hi @alx080 , 

have you setup a python environment according to our requirements.txt (with those exact versions of the dependencies)?

**Comment by kdiers** [comment]
Just a quick notice that we have released version 2.0 of this toolbox, which should solve these issues. If anything persists, please feel free to reopen this issue or write otherwise.
